compiler/optimizer/ExpressionsSimplification.cpp

```diff
@@ -681,18 +681,48 @@ TR_ExpressionsSimplification::removeUncertainBlocks(TR_RegionStructure* region,
             if (trace())
                traceMsg(comp(), "Block_%d is not guaranteed to be executed at least once. Removing it from the list.\n", block->getNumber());
             }
+         else
+            {
+            // If the block is within another loop that's nested in
+            // the current region, remove it from consideration - blocks
+            // must be executed exactly once in the loop to be considered
+            TR_BlockStructure *blockStructure = block->getStructureOf();
+            TR_Structure *parent = blockStructure->getParent();
+
+            while (parent != NULL)
```

Comment:
I wonder if this code could be made simpler if you used the `getContainingLoop` API : https://github.com/eclipse/omr/blob/d067a919eefad16c9ac0c05f8754b1140756549e/compiler/optimizer/Structure.hpp#L104

Reply:
I saw `getContainingLoop`, but I wasn't sure whether that would suffice or checking for `containsInternalCycles` would also be necessary, so I decided to err on the side of caution.  If calling `getContainingLoop` would suffice, I'd be happy to change this.

Reply:
Ah, that is a good point. I don't believe it would suffice, so you can leave it as is